radix-engine/src/engine/call_frame.rs

```diff
@@ -2144,4 +2187,63 @@ where
     fn fee_table(&self) -> &FeeTable {
         self.fee_table
     }
+
+    fn pay_fee(&mut self, vault_id: VaultId, amount: Decimal) -> Result<(), RuntimeError> {
```

Comment:
Is the reason this is here because you can't pay fees from a local vault? If so,
1. Why not?
2. There must be a better place to put this which checks whether the call frame should have access to the vault or not

Reply:
This is to prevent people from draining XRD borrowed through flash loan.